### ISSUE TITLE
Contours: Unhandled Exception Caught: Exception caught with shift contours

### ISSUE THREAD

**Issue by sok0** [CGCookie/retopoflow#833]
The Shift Contours tool doesn't work on my system (Blender 2.82a) but one time it came up for just a moment then it threw this error

Environment:

- RetopoFlow: 3.0.0β
- Blender: 2.82.7 master 2020-03-12
- Platform: Windows, 10, 10.0.18362, AMD64, Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 970/PCIe/SSE2, 4.5.0 NVIDIA 442.50, 4.50 NVIDIA
- Timestamp: 2020-05-21 21:10:31.363518
- Undo: select smart, rotate contours, select smart, select smart, select smart, select smart, select smart, select smart, select smart, select smart

Error Hash: fa418fd6508257c7613a035d4bb662d0

Trace:

- EXCEPTION (<class 'StopIteration'>): 
-     C:\Users\Greg\AppData\Roaming\Blender Foundation\Blender\2.82\scripts\addons\RetopoFlow\retopoflow\rftool_contours\contours.py
- 000 0242:shift_enter() p0 = next(iter(cut.get_origin() for cut in self.move_cuts if cut))
-     .../common\fsm.py
- 001 0054:run() return fn(*args, **kwargs)
```

**Comment by sok0** [comment]
here is part of the file that I am working with, the arm does not allow shift at all, and if I try to shift on the arm then on the other part of the mesh it will give error sometimes, if i don't try to shift on the arm first then the error usually wont show up

[Edit1.zip](https://github.com/CGCookie/retopoflow/files/4666083/Edit1.zip)

**Comment by sok0** [comment]
I tried to see if I could get it to work on the beta2 branch which I know is experimental but it still doesn't work but at least the program is more aware that it's having a problem:

![image](https://user-images.githubusercontent.com/39022834/82632966-b9150a00-9bae-11ea-95c9-12a7981b60e6.png)

**Comment by sok0** [comment]
Automatically Generated Info
-----------------------------------

```
Environment:

- RetopoFlow: 3.0.0β2
- Blender: 2.82.7 master 2020-03-12
- Platform: Windows, 10, 10.0.18362, AMD64, Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 970/PCIe/SSE2, 4.5.0 NVIDIA 442.50, 4.50 NVIDIA
- Timestamp: 2020-05-21 22:08:06.402551
- Undo: 

Error Hash: 65365a48d144e5c2c272f750636851e8

Trace:

- EXCEPTION (<class 'NameError'>): name 'dprint' is not defined
-     C:\Users\Greg\AppData\Roaming\Blender Foundation\Blender\2.82\scripts\addons\retopoflow-b280\retopoflow\rftool_contours\contours.py
- 000 0256:rotateplane_can_enter() dprint('could not crawl around sources for loop')
-     .../common\fsm.py
- 001 0054:run() return fn(*args, **kwargs)
```

**Comment by sok0** [comment]
Okay, the problem is that there is non-manifold geometry on the mesh, some unmerged vertices. I doubt that there is any way to avoid this problem then. But it would be nice if there was a more descriptive error message indicating such. So if this isn't considered a bug, then it is a feature request. thanks.

**Comment by vxlcoder** [comment]
hmmm.  good find.  will take a deeper look at it!

**Comment by vxlcoder** [comment]
OK.  I fixed that exception and I added a bit of code to improve the feedback.

Now, Contours will try to detect when your cut crosses a non-manifold edge but looks like a loop.  If one is detected, a warning shows (this warning can be disabled).  Also added an `O` or a `C` next to the segment counts to indicate if it is a loop (`O`) or a non-loop (`C`).

![image](https://user-images.githubusercontent.com/3790699/83213983-82e90480-a131-11ea-84f2-8429b207ce64.png)

Thanks for your help!
